### PR TITLE
Cherry pick commits from maistra-2.0

### PR DESCRIPTION
**Pull request by luksa**

### CODE REVIEW

**Comment by luksa** [comment]
Don't merge until #16 is merged.

**Review by jwendell — APPROVED** [review]

**Comment by jwendell** [comment]
Please update product side as well, for both 2.0 and 2.1.

**Review by brian-avery — APPROVED** [review]

**Pull request merged**
